rdrf/rdrf/helpers/utils.py

```diff
@@ -263,24 +265,19 @@ def forms_and_sections_containing_cde(registry_model, cde_model_to_find):
 
 def consent_status_for_patient(registry_code, patient):
     from registry.patients.models import ConsentValue
-    from rdrf.models.definition.models import ConsentSection, ConsentQuestion
-
-    consent_sections = ConsentSection.objects.filter(
-        registry__code=registry_code)
-    answers = {}
-    valid = []
-    for consent_section in consent_sections:
-        if consent_section.applicable_to(patient):
-            questions = ConsentQuestion.objects.filter(section=consent_section)
-            for question in questions:
-                try:
-                    cv = ConsentValue.objects.get(
-                        patient=patient, consent_question=question)
-                    answers[cv.consent_question.code] = cv.answer
-                except ConsentValue.DoesNotExist:
-                    pass
-            valid.append(consent_section.is_valid(answers))
-    return all(valid)
+
+    values = ConsentValue.objects.filter(
+        patient=patient,
+        consent_question__section__registry__code=registry_code
+    ).select_related("consent_question", "consent_question__section")
+
+    answers = defaultdict(dict)
+    sections = {}
+    for v in values:
+        section = v.consent_question.section
+        sections[section.code] = section
+        answers[section.code][v.consent_question.code] = v.answer
+    return all(sections[section_code].is_valid(section_answers) for section_code, section_answers in answers.items())
```

Comment:
This will return `True` when the patient has yet to consent, as `values` will be an empty queryset

Reply:
Yes, good catch!
Fixing.